### DISCUSSION TITLE
Is there something to notify my program when the connect client is visible?

### DISCUSSION THREAD

**Discussion by El-Psy-Cangroo** [dtcooper/raspotify#552] [Q&A] [answered]
hi, im making a python program to control the music that is played on raspotify. But the way it looks is that it takes awhile before the raspotify connect shows in spotify or in device list. is there a flag or event that are triggered when its up and going/visible. or do i have to make a loop to check my device list until it pops up?

**Comment by JasonLG1979** [comment, marked as answer]
Not really. Librespot connects as soon as it starts. Any delay you're noticing is on Spotify and/or the app's side. When librespot shows in a client isn't really a concern of librespot.
